### PR TITLE
IBX-154: Allowed to define if the alternative text for image field is required (2.5)

### PR DESCRIPTION
**Pull request by adamwojs**
> JIRA:  [https://issues.ibexa.co/browse/IBX-154](https://issues.ibexa.co/browse/IBX-154)

### CODE REVIEW

File: src/modules/multi-file-upload/services/multi.file.upload.service.js [hunk 1]
```diff
@@ -142,11 +142,23 @@ const prepareStruct = ({ parentInfo, config, languageCode }, data) => {
         .then((response) => response.json())
         .catch(() => window.eZ.helpers.notification.showErrorNotification('Cannot get content type by identifier'))
         .then((response) => {
+            const fileValue = {
+                fileName: data.file.name,
+                data: data.fileReader.result.replace(/^.*;base64,/, ''),
+            }
```
Comment (dew326) [inline src/modules/multi-file-upload/services/multi.file.upload.service.js]:
```suggestion
            };
```

File: src/modules/multi-file-upload/services/multi.file.upload.service.js [hunk 2]
```diff
@@ -142,11 +142,23 @@ const prepareStruct = ({ parentInfo, config, languageCode }, data) => {
...
+
+            if (data.file.type.startsWith('image/')) {
+                fileValue.alternativeText = data.file.name;
+            }
+
             const fields = [
-                { fieldDefinitionIdentifier: mapping.nameFieldIdentifier, fieldValue: data.file.name },
+                {
+                    fieldDefinitionIdentifier: mapping.nameFieldIdentifier,
+                    fieldValue: data.file.name
```
Comment (dew326) [inline src/modules/multi-file-upload/services/multi.file.upload.service.js]: If multiline, the last item should have `,` at the end.

I think it was good with a single line, especially with that the second item also could be a single line now.

File: src/modules/multi-file-upload/services/multi.file.upload.service.js [hunk 3]
```diff
@@ -142,11 +142,23 @@ const prepareStruct = ({ parentInfo, config, languageCode }, data) => {
...
+                },
                 {
                     fieldDefinitionIdentifier: mapping.contentFieldIdentifier,
-                    fieldValue: { fileName: data.file.name, data: data.fileReader.result.replace(/^.*;base64,/, '') },
+                    fieldValue: fileValue
```
Comment (dew326) [inline src/modules/multi-file-upload/services/multi.file.upload.service.js]:
```suggestion
                    fieldValue: fileValue,
```

**Review by dew326 — CHANGES_REQUESTED** [review]

File: src/modules/multi-file-upload/services/multi.file.upload.service.js (hunk 1 above)
Comment (adamwojs) [reply src/modules/multi-file-upload/services/multi.file.upload.service.js]: Fixed in https://github.com/ezsystems/ezplatform-admin-ui-modules/pull/292/commits/a9149aee84ef4eaa608ed4d6d2f288af6c58ef7f

File: src/modules/multi-file-upload/services/multi.file.upload.service.js (hunk 2 above)
Comment (adamwojs) [reply src/modules/multi-file-upload/services/multi.file.upload.service.js]: Reverted to single line in  https://github.com/ezsystems/ezplatform-admin-ui-modules/pull/292/commits/a9149aee84ef4eaa608ed4d6d2f288af6c58ef7f

**Comment by adamwojs** [comment]
PR updated according to code review suggestions.

**Review by dew326 — APPROVED** [review]

**Comment by pbisaga** [comment]
QA approved, checked on ez Platform Enterprise 2.5 with .diff

**Review by pbisaga — APPROVED** [review]

**Pull request merged**